### PR TITLE
8.x 1.x

### PR DESCRIPTION
**Pull request by mccrodp**
Here is my work so far on D8.

Separated the tests out and converted them to be D8 compatible. There are a few unknowns so far, such as the entity_token dependency, which I presume is no longer needed, so it is commented out with a @todo marker.

The tests look like a good start to me, please take a look when you get a chance. They run, but cause all cause a SchemaIncompleteException, probably as they cannot successfully enable the message and message UI modules. I'll come back to these when I convert the module code itself.

### CODE REVIEW

**Comment by mccrodp** [comment]
**So far:**
- Converted all _hook_menu_ entries to routes, including local tasks and contextual links yaml files
- Added a dynamic route entry with associated _MessageRoutes_ class for devel route entries
- Added almost all entries pointing to a controller method as we will perform access control there
- Added Forms at src/Form 

**Todo / Open Issues:**
- Forms and the _MessageUiController_ need a lot of work still, next on my list
- Unsure what the _message_ui_test_ module does really. Will we need it as a module in D8?
- Did you already convert _message_notify_ui_? I seem to recall you mentioning something about this

File: message_ui.links.contextual.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  route_name: message_ui.edit_message
+  group: message-ui
+  title:  'Edit'
+  weight: 0
+
+message_ui.delete_message:
+  route_name: message_ui.delete_message
+  group: message-ui
+  title: 'Delete'
+  weight: 10
```
Comment (RoySegall) [inline message_ui.links.contextual.yml]: New line in the end of the file.

File: message_ui.links.task.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  route_name: message_ui.edit_message
+  base_route: message_ui.show_message
+  title:  'Edit'
+  weight: 0
+
+message_ui.delete_message:
+  route_name: message_ui.delete_message
+  base_route: message_ui.show_message
+  title: 'Delete'
+  weight: 10
```
Comment (RoySegall) [inline message_ui.links.task.yml]: New line in the end of the file.

File: message_ui.routing.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    _controller: '\Drupal\message_ui\Controller\MessageUiController::delete'
+
+message_ui.delete_multiple:
+  path: 'admin/config/development/message_delete_multiple'
+  defaults:
+    _title: 'Message delete multiple'
+    _controller: '\Drupal\message_ui\Controller\MessageUiController::delete_multiple'
+
+route_callbacks:
+  - '\Drupal\message_ui\Routing\ExampleRoutes::routes'
```
Comment (RoySegall) [inline message_ui.routing.yml]: New line in the end of the file.

**Comment by RoySegall** [comment]
First of all 10x for the work. Tag me in any new things in the future because i didn't see this one.

Did you try the module convertor? it can convert menu items and test to Drupal 8. Could be useful.

**Comment by mccrodp** [comment]
@RoySegall thanks for the feedback. 

I had heard of the Drupal Module Upgrader, but as this is my first D8 upgrade I wanted to take a more hands on approach. However, on closer inspection there is an analyze command `dmu-analyze`. I will use this and also compare output of the `dmu-upgrade` command with what I have already, see if I missed anything. 

I'm going to look at this again before next week hopefully and in the meantime I added those "New line in the end of the file" changes to the yaml files and committed the changes.

**Comment by mccrodp** [comment]
I have temporarily added, during the upgrade process, the [html output from running the drupalmoduleupgrader analyze command on the latest version of message_ui](http://opensourceopenmind.ninja/sites/default/files/tmp/upgrade-info.html) on 7.x branch as a reference.

**Comment by mccrodp** [comment]
@RoySegall - I've had a small bit more time for this recently, so have committed some changes. I will have more time for this again soon and also throughout Drupalcon Barcelona I'll work on this whenever I get time.

I've updated `entity_load()` and `user_load()` calls in _message_ui_ and all `l()` and `user_access()` calls. I will leave the _message_notify_ui_ submodule until afterwards. Did you say you had converted this elsewhere already? I'm not sure if I got an answer about that.

There is an [open issue on drupalmoduleupgrader for running the dmu-upgrade](https://www.drupal.org/node/2555203). I'll compare what I have from manual conversion to the auto conversion to D8 when that is resolved.

**Comment by RoySegall** [comment]
Iv'e ported message to to Drupal 8. I'll try to take a look on what youve done so far.

**Comment by mccrodp** [comment]
Yep, I'm referring to your Message for D8 as I go along. I should have a more substantial amount of work to review soon. I've only done about 25% or so of what is needed until I can step through the module code usefully with XDebug.

**Comment by mccrodp** [comment]
A few items I'm currently unsure of now. I will go into more detail later, but just to flag the biggest unknowns at the moment.
- The use of the QueueWorker and associated methods (at the moment I have added some methods to QueueWorker as static methods which I don't think is quite right).
- The best place for the access control checks and associated methods, currently in MessageAccessControlHandler.php

**Comment by mccrodp** [comment]
Hi @RoySegall. I've gotten this quite a bit further now, but I'm running into some more structural unknowns along with the above previously mentioned. For example, I have a central MessageUIController where I redirect my routes to, however it might be better to route directly to newly created Action classes (such as your \Drupal\node\Plugin\Action\DeleteMessage class) or to call my EntityFormController classes directly? https://www.drupal.org/node/1734556

If you get a chance to have even a quick scan of the structure of my route yaml considering the above and let me know which is the preferred approach re Forms, Access, Actions, etc. I would be grateful.

In the meantime I will keep working on it and perhaps make further progress on _message_notify_ui_. Thanks.

**Comment by RoySegall** [comment]
Can you give a commit access so i could do any changes. I'll set up the things and see how it goes.

**Comment by mccrodp** [comment]
Added you as a collaborator to my fork of the project and opened up the Issue tracking on the fork also incase it's useful for separating issues / comments. Thanks.

**Comment by RoySegall** [comment]
10x, i'm looking for now. I'll try to comment today or tomorrow.

**Comment by RoySegall** [comment]
I'm looking on the work but the basic work flow is not working. I can't access the creation page of a message even if i'm admin.

**Comment by mccrodp** [comment]
Hi, yes I know it's not working at all yet. Could you please look at the components in code and let me know what I need to reorganise structurally (from points mentioned in : https://github.com/RoySegall/message_ui/pull/2#issuecomment-139504286) before I can start testing in the UI. I've merely just reorganised to a more D8 structure and converted existing code, but it's not quite right. Once I'm redirected on any major adjustments structurally I'll start going through the UI and get the workflow on track and working correctly. My next step will be stepping through the UI, starting with the message creation page, but any overall comments on the above mentioned in the meantime is much appreciated. Thanks.

**Comment by RoySegall** [comment]
I can't tell if the code is OK or not before the module do something. Start apply the basic work flow and we can the code later.

**Comment by mccrodp** [comment]
No problem, yep, D8 all very new and unknown at the moment :) 

You're right, was just wondering if anything stood out as completely wrong from my work so far. I'll get the basic workflow in place and after that I'll ping you again. Thanks!

**Comment by RoySegall** [comment]
As far as i can see there is not best practice for D8 so we kind off going in the dark for now. Like wise for OG8, we have kind of a strategy.

**Comment by mccrodp** [comment]
True, get it working first and improve towards best practices later, and as you say: it'll be easier to review once there is something working / partially working.

...upgrading OG to D8, that will be an interesting / challenging task! ;)

**Comment by RoySegall** [comment]
Yes, the best minds are on it!

Tag me when you have something or catch me in the near by DrupalCon

**Comment by mccrodp** [comment]
Great, good luck with it!

Yes, hopefully catch you at Drupalcon and if I get a chance to get a good portion of the workflow working in the meantime I'll tag you again.

**Comment by mccrodp** [comment]
Hey @RoySegall - Have a lot more of the workflow in place now and (almost) working.

Main items in front of me to solve now:
- Get <code>MessageViewController</code> or <code>message/{message}</code> route to call _message.html.twig_ correctly when viewing a message; at the moment only the _region.html.twig_ is called
- Get <code>MessageForm</code> to display _message_type_ fields correctly and populate with _message_ values

**Comment by mccrodp** [comment]
Continuing [message_ui port on a separate branch](https://github.com/mccrodp/message_ui/tree/update-d8-stable) until https://github.com/Gizra/message/pull/37 gets accepted into Message module. Made quite a bit of progress. FYI - the reason the fields weren't showing on the form when we were looking at Drupalcon was that they were set to "hidden" in the new "Manage Form Display" UI, haha. Will update on more concrete progress soon.

**Comment by mccrodp** [comment]
@RoySegall - on the above linked branch in https://github.com/RoySegall/message_ui/pull/2#issuecomment-159637555 I have gotten the local tasks working now and have at least the following remaining:
- Ensure entity is properly saved on form Save, displayed on "View" tab and edit form fields are populated correctly on message edit
- Access Control on Message entities similar to that of Node, but simplified
- Port Queue worker functionality for arguments (may need advice on this one)

I'll merge the _message_ui_ work to date into this pull request once https://github.com/Gizra/message/pull/37 or https://github.com/Gizra/message/pull/39 are merged.

**Comment by mccrodp** [comment]
Merged back to main PR branch. Changes made in conjunction with Message module with the following PRs merged together [https://github.com/Gizra/message/pull/39, https://github.com/Gizra/message/pull/48, https://github.com/Gizra/message/pull/50, https://github.com/Gizra/message/pull/51].

_Items remaining:_
1. Get tests running
2. Ensure Access Control is being fired / working
3. Research and fix `MessageUiArgumentsWorker` QueueWorker class
4. Ensure all test are green
5. Release dev release following release of Message

**Comment by mccrodp** [comment]
Good progress on 1. - Tests are now "running" to an extent and travis build file seems to be doing the job: https://travis-ci.org/mccrodp/message_ui/jobs/103951386. Just need to solve the views schema issues in yml file on Message module https://github.com/Gizra/message/issues/52 and get the tests running to completion, even if failing.

**Comment by mccrodp** [comment]
Tests now running due to changes in PR to fix message schema: https://github.com/Gizra/message/pull/53

https://travis-ci.org/mccrodp/message_ui/jobs/104319376

**Comment by mccrodp** [comment]
@RoySegall - Getting much closer with this, not so much left to do:

1 Clean up and correct Access methods
2 Fix Arguments worker

I think I should be ok with 1, but will need to check how the arguments worker functions in D7. Any ideas on this / No. 2 for D8?

At the moment the _.travis.yml_ file points to my branch in this PR (https://github.com/Gizra/message/pull/57) as without valid schema, the tests cannot run / initialise Message module.

Failing due to 1 & 2 above, but running to completion and a lot of green lights on everything else: https://travis-ci.org/mccrodp/message_ui/builds/109325401

**EDIT**: Should those new Message module "owner" getters and setters be in the `MessageInterface` also?

**Comment by mccrodp** [comment]
Hey @RoySegall, any idea what this Exception may mean: 

`'Drupal\Component\Plugin\Exception\PluginNotFoundException: "The "entity:message:foo" plugin does not exist."`

This error is encountered after a call to `drupalPostForm()` to save a message entity  in tests.": https://travis-ci.org/mccrodp/message_ui/jobs/121126540

I tried adding a _SaveMessage_ plugin similar to the plugin in node module, but no luck. https://github.com/mccrodp/message_ui/blob/8.x-1.x/src/Plugin/Action/SaveMessage.php

Tried asking a few times on IRC, but I'm a bit stuck with it. Any help appreciated!!

**Comment by mccrodp** [comment]
Also, what's the best way to merge this PR? It's quite big, should we have smaller PRs, or will you just merge when it gets closer to completion as there is no 8.x branch to merge to at the moment. Thanks @RoySegall

**Comment by mccrodp** [comment]
I opened issues for the remaining tasks: https://github.com/mccrodp/message_ui/issues

**Comment by mccrodp** [comment]
@RoySegall - Some _message_ui_ goals for this week: https://github.com/mccrodp/message_ui/wiki/Development-Wiki

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  # Ensure we have the latest sources.
+  - sudo apt-get -y update
+
+  # Composer.
+  - sed -i '1i export PATH="$HOME/.composer/vendor/bin:$PATH"' $HOME/.bashrc
+  - source $HOME/.bashrc
+  # - composer self-update
+
+  # Drush.
+  - composer global require "youngj/httpserver:dev-master#41dd2b7"
```
Comment (RoySegall) [inline .travis.yml:33]: We might want to use 8.0.5 version.

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  - php $TRAVIS_BUILD_DIR/drupal/core/scripts/run-tests.sh --verbose --color --php `which php` --url http://127.0.0.1:8080 "$MODULE_TEST_GROUP" | tee /tmp/test.txt; TEST_EXIT=${PIPESTATUS[0]}; echo $TEST_EXIT
+  # Check if we had fails in the run-tests.sh script
+  # Exit with the inverted value, because if there are no fails found, it will
+  # exit with 1 and for us that is a good thing so invert it to 0. Travis has
+  # some issues with the exclamation mark in front so we have to fiddle a bit.
+  # Also make the grep case insensitive and fail on run-tests.sh regular fails
+  # as well on fatal errors.
+  - TEST_OUTPUT=$(! egrep -i "([0-9]+ fails)|(PHP Fatal error)|([0-9]+ exceptions)" /tmp/test.txt > /dev/null)$?; echo $TEST_OUTPUT
+  # Exit the build
+  - if [ $TEST_EXIT -eq 0 ] && [ $TEST_OUTPUT -eq 0 ]; then exit 0; else exit 1; fi
```
Comment (RoySegall) [inline .travis.yml]: missing line in the end of the file

File: config/install/message_ui.settings.yml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+update_tokens:
+  update_tokens: false
+  how_to_act: 'update_when_added'
+  number_items: 250
+show_preview: true
+anonymous: 'Anonymous' # What is this variable and how best to store/access it?
```
Comment (RoySegall) [inline config/install/message_ui.settings.yml]: If we are not using setting any where so let's remove it.

File: config/install/system.action.message_save_action.yml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+langcode: en
+status: true
+dependencies:
+  module:
+    - message
+id: message_save_action
+label: 'Save message'
+type: message
+plugin: message_save_action
+configuration: {  }
```
Comment (RoySegall) [inline config/install/system.action.message_save_action.yml]: Missing line in the end of the file.

File: message_notify_ui/message_notify_ui.info.yml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+name: 'Message Notify UI'
+type: module
+description: 'Integrating the message notify into the UI functionality.'
+core: 8.x
+package: Message
+
+dependencies:
+ - views
+ - views_bulk_operations
```
Comment (RoySegall) [inline message_notify_ui/message_notify_ui.info.yml]: I don't think the vbo module live as it is.

File: message_notify_ui/src/Plugin/Views/field/NotifyButton.php [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+  /**
+   * {@inheritdoc}
+   */
+  public function render(ResultRow $values) {
+    $message = Message::load($values->_entity->id());
+
+    if (user_access('send message notify')) {
+      return l(t('Notify'), 'message/' . $message->id() . '/notify');
+    }
+    return NULL;
```
Comment (RoySegall) [inline message_notify_ui/src/Plugin/Views/field/NotifyButton.php:44]: Just return. No need for NULL.

File: message_ui.js [hunk 10]
```diff
@@ -1,9 +1,9 @@
-(function ($) {
+(function ($, Drupal, drupalSettings) {
```
Comment (RoySegall) [inline message_ui.js]: Let's move it back to the js folder.

File: message_ui.libraries.yml [hunk 11]
```diff
@@ -0,0 +1,9 @@
+message_ui.message:
+  version: VERSION
+  js:
+    message_ui.js: {}
+  dependencies:
+    - core/jquery
+    - core/drupal
+    - core/drupalSettings
+    - core/drupal.form
```
Comment (RoySegall) [inline message_ui.libraries.yml]: Missing line in the end of the file.

File: message_ui.links.action.yml [hunk 12]
```diff
@@ -0,0 +1,6 @@
+# All action links for this module
+message_ui.add_page:
+  route_name: message_ui.add_page
+  title: 'Add message'
+  appears_on:
+    - view.message.page_1
```
Comment (RoySegall) [inline message_ui.links.action.yml]: Missing line in the end of the file.

File: message_ui.links.contextual.yml [hunk 13]
```diff
@@ -0,0 +1,10 @@
+entity.message.edit_form:
+  route_name: entity.message.edit_form
+  group: message-ui
+  title:  'Edit'
+  weight: 0
+entity.message.delete_form:
+  route_name: entity.message.delete_form
+  group: message-ui
+  title: 'Delete'
+  weight: 10
```
Comment (RoySegall) [inline message_ui.links.contextual.yml]: Missing line in the end of the file.

File: message_ui.links.task.yml [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+message.page_edit:
+  route_name: entity.message.edit_form
+  base_route: entity.message.canonical
+  title: Edit
+
+message.delete_confirm:
+  route_name:  entity.message.delete_form
+  base_route:  entity.message.canonical
+  title: Delete
+  weight: 10
```
Comment (RoySegall) [inline message_ui.links.task.yml]: Missing line in the end of the file.

File: message_ui.module [hunk 15]
```diff
@@ -1,447 +1,245 @@
...
-      $items['admin/content/message/create/' . str_replace('_', '-', $type)] = array(
-        'title' => ucfirst(str_replace('_', ' ', $title)),
-        'description' => 'Create a new message' . $title . ' instance',
-        'page callback' => 'drupal_get_form',
-        'page arguments' => array('message_ui_instance_message_manage', $type),
-        'access arguments' => array('create', $type),
-        'access callback' => 'message_ui_access_control',
-      );
+function message_ui_entity_base_field_info_alter(&$fields, EntityTypeInterface $entity_type) {
+    // Alter the uid and created field to include display settings.
```
Comment (RoySegall) [inline message_ui.module]: what the use of this one?

File: message_ui.module [hunk 16]
```diff
@@ -1,447 +1,245 @@
...
-      foreach ($types as $type => $title) {
-        $permissions[$operation . ' a ' . $type . ' message instance'] = array(
-          'title' => t(ucfirst($operation) . ' a new message instance for ' . $title),
-          'description' => t('Allowing to ' . $operation . ' an instance for the ' . $title . ' message type'),
-        );
-      }
+function message_ui_entity_type_alter(array &$entity_types) {
+    if (isset($entity_types['message'])) {
+
+        /**
```
Comment (RoySegall) [inline message_ui.module]: The indentation looks wrong.

If this one commented out then remove it.

File: message_ui.module [hunk 17]
```diff
@@ -1,447 +1,245 @@
...
  * Implements hook_theme().
  */
 function message_ui_theme() {
-  return array(
-    'message_ui_create_message' => array(
-      'variables' => array('items' => NULL),
-    ),
-  );
+    return array(
+        'message_add_list' => array(
```
Comment (RoySegall) [inline message_ui.module]: Indentation looks wrong.

File: message_ui.module [hunk 18]
```diff
@@ -1,447 +1,245 @@
...
-    $output .= '<li class="clearfix">';
-    $output .= '<span class="label">' . l(ucfirst(str_replace('_', ' ', $item['type'])), 'admin/content/message/create/' . str_replace('_', '-', $item['type'])) . '</span>';
-    $output .= '<div class="description">' . t('Create a message instance of @type', array('@type' => $item['name'])) . '</div>';
-    $output .= '</li>';
-  }
-
-  $output .= '</ul>';
-
-  return $output;
+function template_preprocess_message_add_list(&$variables) {
```
Comment (RoySegall) [inline message_ui.module:115]: Indentation looks wrong here as well.

File: message_ui.module [hunk 19]
```diff
@@ -1,447 +1,245 @@
...
 
 /**
- * Display list of message types to create an instance for them.
+ * Implements hook_form_FORM_ID_alter().
  */
-function message_ui_create_new_message_instance_list() {
-  $items = array();
-  $allowed_types = message_ui_user_can_create_message();
+function message_ui_form_message_user_admin_settings_alter(&$form, \Drupal\Core\Form\FormStateInterface $form_state) {
+    $form['update_tokens'] = array(
```
Comment (RoySegall) [inline message_ui.module]: Indentation looks wrong here as well.

File: message_ui.module [hunk 20]
```diff
@@ -454,443 +252,92 @@ function message_ui_form_message_user_admin_settings_alter(&$form, $form_state)
...
+    if ($type != 'message_type') {
+        return FALSE;
+    }
 
-  foreach ($arguments as $token) {
-    // Get the hard coded value of the message and him in the message.
-    $token_name = str_replace(array('@{', '}'), array('[', ']'), $token);
-    $value = token_replace($token_name, array('message' => $message));
+    $query = \Drupal::entityQuery('message');
+    $result = $query
```
Comment (RoySegall) [inline message_ui.module]: Indentation

File: message_ui_test/message_ui_test.info.yml [hunk 21]
```diff
@@ -0,0 +1,10 @@
+name: 'Message UI test'
+type: module
+description: 'Dummy module for checking how other modules will interact with message UI'
+core: 8.x
+package: Message
+
+dependencies:
+ - message
+ - views
+ - message_ui
```
Comment (RoySegall) [inline message_ui_test/message_ui_test.info.yml:10]: Missing line in the end of the file.

File: src/Controller/MessageController.php [hunk 22]
```diff
@@ -0,0 +1,106 @@
...
+    // Bypass the admin/content/messages/create listing if only one content type is available.
+    if (count($content) == 1) {
+      $type = array_shift($content);
+      return $this->redirect('message_ui.add', array('message_type' => $type->id()));
+    }
+
+    $build['#content'] = $content;
+
+    if (!empty($build)) {
+      return $build;
```
Comment (RoySegall) [inline src/Controller/MessageController.php]: You can remove the `$build` array and do some thing like:

``` php
return [
  ...
];
```

And not declare the variable.

File: src/Entity/MessageRouteProvider.php [hunk 23]
```diff
@@ -0,0 +1,43 @@
...
+        '_entity_form' => 'message.delete',
+        '_title' => 'Delete',
+      ])
+      ->setRequirement('_message_ui_access', 'delete')
+      ->setOption('_admin_route', TRUE);
+    $route_collection->add('entity.message.delete_form', $route);
+
+    return $route_collection;
+  }
+}
```
Comment (RoySegall) [inline src/Entity/MessageRouteProvider.php]: Missing line in the end of the file.

File: src/Form/DeleteMultiple.php [hunk 24]
```diff
@@ -0,0 +1,189 @@
...
+          '@type' => $message->getType()->label(),
+        );
+        return t('Delete message ID @id fo type @type', $params);
+      }, $this->messages),
+    );
+    $form = parent::buildForm($form, $form_state);
+
+    $form['actions']['cancel']['#href'] = $this->getCancelRoute();
+
+    // @todo - from message_ui in D7.
```
Comment (RoySegall) [inline src/Form/DeleteMultiple.php]: If commented out then remove it.

File: src/QueueWorker/MessageArgumentsWorker.php [hunk 25]
```diff
@@ -0,0 +1,124 @@
...
+    $messages = Message::loadMultiple($mids);
+
+    foreach ($messages as $message) {
+      /* @var Message $message */
+      MessageArgumentsWorker::messageArgumentsUpdate($message, $arguments);
+    }
+  }
+}
+
+?>
```
Comment (RoySegall) [inline src/QueueWorker/MessageArgumentsWorker.php]: No need for the close tag and there is a missing line in the end of the file.

File: src/Routing/MessageRoutes.php [hunk 26]
```diff
@@ -0,0 +1,66 @@
...
+ * Defines dynamic routes.
+ */
+class MessageRoutes {
+  /**
+   * {@inheritdoc}
+   */
+  public function routes() {
+    $routes = array();
+
+    // @todo Complete code for interaction with devel module for this menu item.
```
Comment (RoySegall) [inline src/Routing/MessageRoutes.php]: if it's commented out let's remove it.

File: src/Tests/MessageUiHardCodedArguments.php [hunk 27]
```diff
@@ -0,0 +1,111 @@
...
+
+    // Update the message arguments manually.
+    $edit = array(
+      'name' => $this->user2->getUsername(),
+      'replace_tokens' => 'update_manually',
+      '@{message:user:name}' => 'Dummy name',
+    );
+    $this->drupalPostForm('message/' . $message->id() . '/edit', $edit, t('Update'));
+    $this->assertText('Dummy name', 'The hard coded token was updated with a custom value.');
+  }
```
Comment (RoySegall) [inline src/Tests/MessageUiHardCodedArguments.php:113]: An empty line is needed between the last method and the closure of the message.

File: src/Tests/MessageUiMassiveHardCodedArguments.php [hunk 28]
```diff
@@ -0,0 +1,117 @@
...
+/**
+ * Testing the update of the hard coded arguments in massive way.
+ *
+ * @group Message UI
+ */
+class MessageUiMassiveHardCodedArguments extends MessageTestBase {
+
+  /**
+   * The user object.
+   * @var
```
Comment (RoySegall) [inline src/Tests/MessageUiMassiveHardCodedArguments.php:22]: Add a line and specify the type of the property. it's probably `AccountInterface` or something similar.

File: config/install/message_ui.settings.yml (hunk 6 above)
Comment (mccrodp) [inline config/install/message_ui.settings.yml]: That is a comment I added at the start, I'll remove the comment. We use this setting at: `/src/Form/MessageForm.php:57`

File: .travis.yml (hunk 5 above)
Comment (mccrodp) [inline .travis.yml]: Done

File: .travis.yml (hunk 4 above)
Comment (mccrodp) [inline .travis.yml:33]: Done.

File: message_ui.module (hunk 15 above)
Comment (mccrodp) [inline message_ui.module]: @RoySegall It allows the "Authoring Information" or "Owner Information" tab in the vertical tabs to show on the entity form. In node this is in Node.php, but we either have to put this in Message.php or here in an alter hook.

**Comment by mccrodp** [comment]
Hi @RoySegall - I have added the step to show fields to the README. I have made the remaining code review changes also.

Re: showing message text, I was using the default class `EntityViewBuilder` so I changed it to `MessageViewBuilder`, however I cannot see the partials being displayed. The build array is generated by the Message module, it includes the partials. I am not sure if this working.

Can we merge, then I will look at this in the Message module and make a PR on message or alter on Message UI, whichever is appropriate separately to this. Be nice to get close this big ugly PR :) Thanks!

File: message_ui.module (hunk 15 above)
Comment (RoySegall) [inline message_ui.module]: OK. It seems we have indentation issues here.

File: message_ui.module [hunk 29]
```diff
@@ -1,447 +1,226 @@
...
-
-    if ($types = message_ui_get_types()) {
-      foreach ($types as $type => $title) {
-        $permissions[$operation . ' a ' . $type . ' message instance'] = array(
-          'title' => t(ucfirst($operation) . ' a new message instance for ' . $title),
-          'description' => t('Allowing to ' . $operation . ' an instance for the ' . $title . ' message type'),
-        );
-      }
+function message_ui_entity_type_alter(array &$entity_types) {
+    if (isset($entity_types['message'])) {
```
Comment (RoySegall) [inline message_ui.module]: Indentation issues here as well.

File: message_ui.module [hunk 30]
```diff
@@ -454,443 +233,92 @@ function message_ui_form_message_user_admin_settings_alter(&$form, $form_state)
...
-}
+    if ($how_update == 'update_with_batch') {
 
-/**
- * Validate the submitted message.
- */
-function message_ui_instance_message_manage_validate($form, &$form_state) {
-  field_attach_form_validate('message', $form_state['#entity'], $form, $form_state);
-}
+        // Get all the messages.
```
Comment (RoySegall) [inline message_ui.module]: Indentation issues over here.

File: src/Entity/MessageRouteProvider.php [hunk 31]
```diff
@@ -0,0 +1,43 @@
...
+class MessageRouteProvider implements EntityRouteProviderInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getRoutes( EntityTypeInterface $entity_type) {
+    $route_collection = new RouteCollection();
+
+    $route = (new Route('/message/{message}/edit'))
+        ->setDefault('_entity_form', 'message.edit')
```
Comment (RoySegall) [inline src/Entity/MessageRouteProvider.php]: Indentation issues over here

**Comment by RoySegall** [comment]
@mccrodp It seems that you have indentation issues. Please check they are set to 2 spaced in you IDE.

**Comment by mccrodp** [comment]
@RoySegall - done, I committed reformatted code. I had set it in the IDE, but had to go through each file and reformat individually :)

File: .travis.yml (hunk 4 above)
Comment (RoySegall) [inline .travis.yml:33]: Can't see the changes over this one.

File: config/install/message_ui.settings.yml [hunk 32]
```diff
@@ -0,0 +1,6 @@
+update_tokens:
+  update_tokens: false
+  how_to_act: 'update_when_added'
+  number_items: 250
+show_preview: true
+anonymous: 'Anonymous'
```
Comment (RoySegall) [inline config/install/message_ui.settings.yml]: Missing line in the end of the file.

**Comment by RoySegall** [comment]
I'll merge the branch once I'll apply the small fixes.

File: .travis.yml (hunk 4 above)
Comment (mccrodp) [inline .travis.yml:33]: See line 20 & 63:

https://github.com/mccrodp/message_ui/blob/8.x-1.x/.travis.yml#L20
https://github.com/mccrodp/message_ui/blob/8.x-1.x/.travis.yml#L63

**Comment by mccrodp** [comment]
@RoySegall - added the new line and remove a comment from travis, so the `composer install` is line 62 now.

**Comment by RoySegall** [comment]
Giving this another look there is no messages to create:
add_message___site-install_and_develeventsubscriber_php_-_drupal8_-___applications_mamp_htdocs_drupal8_

**Comment by mccrodp** [comment]
@RoySegall Thank you. I had forgotten to add the `#theme` to the build array since I had removed that local `$build` variable. Fixed now.

**Comment by RoySegall** [comment]
I testes it again and when I created a message comment i got a fata error.

**Comment by mccrodp** [comment]
Really? What's the error? I'm using it no problem with Message Private module's private_message type at moment. I do have an open issue on Message module's MessageViewBuilder at the moment though, was it that? Also a PR on Message I am using with this module, might be related.

I'll take a look with comment message type soon and see. Some day we will get this merged :)

**Comment by mccrodp** [comment]
Hi @RoySegall I didn't look at this exactly as the work on renaming Message Type to Message Template was merged.

I renamed all to "message template" and I was able to add, edit, delete a message via message_ui.

[I couldn't check with message_example due: https://github.com/Gizra/message/issues/114]

**Comment by mccrodp** [comment]
Hi @RoySegall It would be great to get this merged whenever you get a chance. Then we can start working on things in more manageable chunks / PRs. 

I've had people messaging me asking about a dev release for 8.x. It would be best to get people to test this now that Message has a dev release on Drupal.org. Thanks!

**Comment by RoySegall** [comment]
I saw you push more work. I'll have a look on this and if everything is OK i'll merge.

**Comment by mccrodp** [comment]
@RoySegall - sounds good Roy. There will be some work remaining I'm sure, but if nothing major we can make some requirements toward a dev release after merge and I can create separate PRs against each one! 👯 What do you think?

**Comment by RoySegall** [comment]
OK. I pushed the `8.x-1.x`. There are some [issues](https://github.com/RoySegall/message_ui/issues?q=is%3Aissue+is%3Aopen+label%3A%22Drupal+8%22) before i'll create an alpha release since this one relate to UX of the user.

Except that there are some coding standard issues which can be detect by the code sniffer. I'll add an issue and we can apply the same action as done in OG. 

God job!

**Comment by mccrodp** [comment]
Thanks @RoySegall - that's fantastic. Yes there are certainly more issues to work on here, I'll create separate PRs now. Will you push a dev release to drupal.org in the meantime?

I have a few issues on Message that I want to work on now that this is merged also. So I'll ping you on those if I need some input.

Shall we close this one?

**Comment by RoySegall** [comment]
No. Until the other issues we won't create a release.

**Pull request closed without merging**